packages/pluggableWidgets/video-player-native/src/VideoPlayer.tsx

```diff
@@ -36,27 +53,129 @@ export function VideoPlayer(props: VideoPlayerProps<VideoStyle>): ReactElement {
+            return;
+        }
+        if (!showControls) {
+            showControlsHandler();
+        } else {
+            clearTimeout(timeoutRef.current as NodeJS.Timeout);
+            setShowControls(false);
+        }
+    }, [props.showControls, showControls]);
+
+    function fullScreenHandler(isFullScreen: boolean): void {
+        setFullScreen(isFullScreen);
+        StatusBar.setHidden(isFullScreen);
+    }
+
+    const videoProps: VideoProperties = {
+        testID: props.name,
+        source: { uri: isAvailable(props.videoUrl) ? props.videoUrl.value : undefined },
+        muted: props.muted,
+        repeat: props.loop,
+        controls: props.showControls,
+        onLoadStart: () => setStatus(StatusEnum.LOADING),
+        onError: () => setStatus(StatusEnum.ERROR),
+        useTextureView: false,
+        resizeMode: props.aspectRatio ? "contain" : "stretch",
+        onProgress: ({ currentTime }: OnProgressData) => setCurrentPlayTime(currentTime)
+    };
+
+    const isAndroid = Platform.OS === "android";
+    const errorText = status === StatusEnum.ERROR && <Text style={styles.errorMessage}>The video failed to load</Text>;
+
     return (
-        <View style={styles.container}>
-            {status === StatusEnum.LOADING && <ActivityIndicator color={styles.indicator.color} size="large" />}
-            {status === StatusEnum.ERROR && <Text style={styles.errorMessage}>The video failed to load :(</Text>}
-            <Video
-                testID={props.name}
-                source={{ uri: isAvailable(props.videoUrl) ? props.videoUrl.value : undefined }}
-                paused={!props.autoStart}
-                muted={props.muted}
-                repeat={props.loop}
-                controls={props.showControls}
-                onLoadStart={() => setStatus(StatusEnum.LOADING)}
-                onLoad={data => {
-                    setStatus(StatusEnum.READY);
-                    setVideoAspectRatio(data.naturalSize.width / data.naturalSize.height);
-                }}
-                onError={() => setStatus(StatusEnum.ERROR)}
-                style={status !== StatusEnum.READY ? { height: 0 } : styles.video}
-                useTextureView={false}
-                resizeMode={props.aspectRatio ? "contain" : "stretch"}
-            />
-        </View>
+        <Fragment>
+            {isAndroid && (
+                <Modal
+                    isVisible={fullScreen}
+                    style={styles.fullScreenVideoPlayer}
+                    onBackButtonPress={() => fullScreenHandler(false)}
+                    testID="fullscreen-modal"
+                >
+                    <TouchableWithoutFeedback onPress={onVideoPressHandler} testID="fullscreen-overlay">
+                        <Video
+                            {...videoProps}
+                            ref={fullScreenPlayerRef}
+                            paused={false}
+                            onLoad={data => {
```

Comment:
I suggest using `useCallback` for performance

Reply:
On second thought this may not be helpful. Best to check. Let me know what you think

Reply:
as we discussed after looking at the implementation, using `useCallback` here will not be our best option.